addons/website_sale_stock/static/src/js/variant_mixin.js

```diff
@@ -61,7 +62,8 @@ VariantMixin._onChangeCombinationStock = function (ev, $parent, combination) {
             $addQtyInput.val(qty);
         }
         if (combination.free_qty < 1) {
-            $parent.find('#add_to_cart').addClass('disabled out_of_stock');
+            addToCartWrapper.classList.replace('d-inline', 'd-none');
```

Comment:
in the current PR, this is broken for variants even when updating the module

scenario: select a variant without stock => button is hidden ; select a variant with stock => button is not shown back

you need to possibly show the add_to_cart possible back when you change variant, in the current code, this is handled by doing above:

```js
    $parent.find('#add_to_cart').removeClass('out_of_stock');
    $parent.find('.o_we_buy_now').removeClass('out_of_stock');
```
and the `disabled` class are removed in these lines in website_sale (called inside `_onChangeCombination`):

https://github.com/odoo/odoo/blob/276eb0192fdddb736453857c18bf9f0cccecb4a3/addons/website_sale/static/src/js/website_sale.js#L700-L701